gcip/addons/container/jobs/crane.py

```diff
@@ -51,46 +52,65 @@ def copy(
+    image_name: Optional[str] = None,
+    image_tag: Optional[str] = None,
     docker_client_config: Optional[DockerClientConfig] = None,
     crane_image: Optional[Union[Image, str]] = None,
 ) -> Job:
     """
     Creates a job to push container image to remote container registry with `crane`.
 
     The image to copy must be in a `tarball` format. It gets validated with crane
-    and is pushed to `dst` destination registry.
+    and is pushed to `dst_registry` destination registry.
 
     Args:
-        dst (str): Registry URL to copy container image to.
-        image_path (Optional[str], optional): Path where to find the container image.
+        dst_registry (str): Registry URL to copy container image to.
+        tar_path (Optional[str], optional): Path where to find the container image tarball.
             If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_DIR`. Defaults to None.
         image_name (Optional[str], optional): Container image name, searched for in `image_path` and gets `.tar` appended.
             If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_NAME`. Defaults to None.
+        image_tag (Optional[str]): The tag the image will be tagged with.
+            Defaults to `PredefinedVariables.CI_COMMIT_REF_SLUG` or `PredefinedVariables.CI_COMMIT_TAG`.
         docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
-            used by crane to authenticate against given registries. Defaults to None.
+            to authenticate against given registries. Defaults to a `DockerClientConfig` with login to the official Docker Hub
+            and expecting credentials given as environment variables `REGISTRY_USER` and `REGISTRY_LOGIN`.
+
         crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
             Defaults to PredefindedImages.CRANE.
 
     Returns:
         Job: Returns a `gcip.Job`, with neccessary configuration to push a container image stored as a `tarball` to a remote registry.
+            Job runs in ```namespace=push```
     """
     if not crane_image:
         crane_image = PredefinedImages.CRANE
 
-    if not image_path:
-        image_path = PredefinedVariables.CI_PROJECT_DIR
+    if not tar_path:
+        tar_path = PredefinedVariables.CI_PROJECT_DIR
     if not image_name:
         image_name = PredefinedVariables.CI_PROJECT_NAME
+    image_path = image_name.replace("/", "_")
+
+    if not docker_client_config:
+        docker_client_config = DockerClientConfig()
+        docker_client_config.add_auth(registry=Registry.DOCKER)
+
+    if not image_tag:
+        if PredefinedVariables.CI_COMMIT_TAG:
+            image_tag = PredefinedVariables.CI_COMMIT_TAG
+        elif PredefinedVariables.CI_COMMIT_REF_SLUG:
+            image_tag = PredefinedVariables.CI_COMMIT_REF_SLUG
+    image_tag_postfix = f":{image_tag}"
```

Comment:
What if none of the conditions match? Than `image_tag_postix = ":None"`- isn't it?

Reply:
Not using `elif` would express what you want.
(Don't know how to create a multiline suggestion here)

```
    if not image_tag:
        if PredefinedVariables.CI_COMMIT_TAG:
            image_tag = PredefinedVariables.CI_COMMIT_TAG
        else PredefinedVariables.CI_COMMIT_REF_SLUG:
            image_tag = PredefinedVariables.CI_COMMIT_REF_SLUG
```

Reply:
Yes this is a better solution. I will add this.